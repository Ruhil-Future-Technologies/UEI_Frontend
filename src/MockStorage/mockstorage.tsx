/* eslint-disable @typescript-eslint/no-explicit-any */
import { Box } from '../Pages/AcademicHistory/AcademicHistory';

export const mockStorage: any = {
  menulist1: JSON.stringify([
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 23,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Menu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 56,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 23,
          menu_master_name: 'User Authorization',
          menu_name: 'Menu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 24,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Submenu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 57,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 24,
          menu_master_name: 'User Authorization',
          menu_name: 'Submenu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
  ]),
};

export const contextValue = {
  namepro: { id: 73, user_type: 'admin', userid: 'ashish7080@gmail.com' },
  setNamepro: jest.fn(),
  logoutpro: jest.fn(),
  setProImage: jest.fn(),
  proImage: '',
  setProPercentage: jest.fn(),
  ProPercentage: '',
  setNamecolor: jest.fn(),
  namecolor: '#ffffff',
  activeForm: 0, // Assign a number value, e.g., 0
  setActiveForm: jest.fn(), // Mock function remains unchanged
};

export const mockBoxes: Box[] = [
  {
    id: 334,
    institute_type: 'college',
    board: '',
    state_for_stateboard: '',
    institute_id: '95',
    course_id: 87,
    learning_style: 'online',
    class_id: '24',
    year: '2015-12-31T18:30:00.000Z',
    stream: '',
    university_id: '',
    sem_id: 1,
  },
];

export const mockClassValue = {
  data: [
    {
      class_name: 'class_12',
      created_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      created_by: 'afasfsafsaf',
      id: '5500819c-09e5-4543-8cf2-8f0d5cf6ffc3',
      is_active: true,
      is_deleted: false,
      updated_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      updated_by: 'Admin',
    },
  ],
  status: 200,
  message: 'Class Found Successfully',
};

export const mockbasicinfoPayload = {
  payload: [
    {
      student_login_id: '123',
      first_name: 'John',
      last_name: 'Patel',
      gender: 'male',
      dob: '01/01/2024',
      father_name: 'father',
      mother_name: 'mother',
      guardian_name: 'guardian',
      pic_path: 'pic_path',
      aim: 'aim',
    },
  ],
};
