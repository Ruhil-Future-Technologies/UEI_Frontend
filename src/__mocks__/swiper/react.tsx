import React from 'react';

export const Swiper: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;

export const SwiperSlide: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;
