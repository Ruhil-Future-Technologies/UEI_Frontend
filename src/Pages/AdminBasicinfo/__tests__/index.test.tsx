import { fireEvent, render } from "@testing-library/react";
import React from "react";
import AdminBasicInfo from "..";
import NameContext from "../../Context/NameContext";
import { contextValue } from "../../../MockStorage/mockstorage";
import { MemoryRouter } from "react-router-dom";
import dayjs from "dayjs";

describe("Admin Basic info Component", () => {
  // Mocking the required props
  const mockSetActiveForm = jest.fn();
  const mockActiveForm = 1; // Dummy value

  const renderedComponent = () => {
    return render(
      <NameContext.Provider value={contextValue}>
        <MemoryRouter>
          <AdminBasicInfo
            setActiveForm={mockSetActiveForm}
            activeForm={mockActiveForm}
          />
        </MemoryRouter>
      </NameContext.Provider>
    );
  };

  it("should render admin info fields correctly", () => {
    const { getByTestId, getByLabelText } = renderedComponent();
    const datePicker = getByLabelText("datepicker_label") as HTMLInputElement;

    expect(getByTestId("first_name")).toBeInTheDocument();
    expect(getByTestId("last_name")).toBeInTheDocument();
    expect(getByTestId("gender")).toBeInTheDocument();
    expect(datePicker).toBeInTheDocument();
    expect(getByTestId("father_name")).toBeInTheDocument();
    expect(getByTestId("mother_name")).toBeInTheDocument();
    expect(getByTestId("guardian_name")).toBeInTheDocument();
    expect(getByTestId("department_name")).toBeInTheDocument();
    expect(getByTestId("profile_image")).toBeInTheDocument();
    expect(getByTestId("next_button")).toBeInTheDocument();
  });

  it("should call handleInputChange function on input change", () => {
    const { getByTestId } = renderedComponent();

    // Find the input element by test ID
    const firstNameInput = getByTestId("first_name") as HTMLInputElement;
    const lastNameInput = getByTestId("last_name") as HTMLInputElement;

    // Get the male radio button and cast it to HTMLInputElement
    const gender = getByTestId("gender").querySelector(
      'input[value="male"]'
    ) as HTMLInputElement; // Cast to HTMLInputElement

    const father_name = getByTestId("father_name") as HTMLInputElement;
    const mother_name = getByTestId("mother_name") as HTMLInputElement;
    const guardian_name = getByTestId("guardian_name") as HTMLInputElement;

    // Simulate an onChange event
    fireEvent.change(firstNameInput, { target: { value: "John" } });
    fireEvent.change(lastNameInput, { target: { value: "Patel" } });

    // Ensure gender is not null and simulate a click event on the male radio button
    if (gender) {
      fireEvent.click(gender);
    }

    fireEvent.change(father_name, { target: { value: "father" } });
    fireEvent.change(mother_name, { target: { value: "mother" } });
    fireEvent.change(guardian_name, { target: { value: "guardian" } });

    // Check if the first_name value has been updated
    expect(firstNameInput.value).toBe("John");
    expect(lastNameInput.value).toBe("Patel");

    // Check if the gender radio button is selected
    if (gender) {
      expect(gender.checked).toBe(true); // Check the 'checked' property after firing the click
    }

    expect(father_name.value).toBe("father");
    expect(mother_name.value).toBe("mother");
    expect(guardian_name.value).toBe("guardian");
  });

  it("should call handleDateChange when DatePicker value changes", async () => {
    const { getByLabelText } = renderedComponent();

    const datePicker = getByLabelText("datepicker_label") as HTMLInputElement;
    const zeroPad = (num: number) => String(num).padStart(2, "0");
    const dayString = zeroPad(1);
    const monthString = zeroPad(1);
    const yearString = String(2001);
    const dateText = `${monthString}/${dayString}/${yearString}`;
    fireEvent.change(datePicker, { target: { value: dateText } });
    const datePickerValue = `${monthString}/${dayString}/${yearString}`;
    expect(datePicker.value).toBe(datePickerValue);
  });

  it("should call handleDateChange and give errors when DatePicker value changes for future dates or invalid dates", async () => {
    const { getByLabelText, getByText } = renderedComponent();

    // Get the date picker input
    const datePicker = getByLabelText("datepicker_label");

    // Test with a future date (future dates should show an error)
    const futureDate = dayjs().add(1, "year").format("DD/MM/YYYY");
    fireEvent.change(datePicker, { target: { value: futureDate } });

    // After changing to a future date, we expect the error message for future dates
    expect(getByText("Future dates are not allowed.")).toBeInTheDocument();

    // Test with a date where the person is younger than 6 years (should show age error)
    const youngerThanSixDate = dayjs()
      .subtract(5, "years")
      .format("DD/MM/YYYY");
    fireEvent.change(datePicker, { target: { value: youngerThanSixDate } });

    // After changing to a date where the person is younger than 6 years, we expect the age error
    expect(getByText("You must be at least 6 years old.")).toBeInTheDocument();
  });

  it("should update file preview when a valid image is selected", async () => {
    // Render the component
    const { getByTestId, findByAltText } = renderedComponent();

    // Get the file input element
    const fileInput = getByTestId("profile_image") as HTMLInputElement;

    // Create a mock file with a valid image type and size under 3MB
    const file = new File(["image content"], "valid-image.png", {
      type: "image/png",
    });

    // Create a mock of FileReader to simulate the onloadend event
    const fileReaderMock = {
      onloadend: jest.fn(),
      readAsDataURL: jest.fn(),
      result: "data:image/png;base64,exampledata", // Mocked file preview URL
      EMPTY: 0,
      LOADING: 1,
      DONE: 2,
    };

    // Override global FileReader with the mock implementation
    Object.defineProperty(global, "FileReader", {
      writable: true,
      value: jest.fn().mockImplementation(() => fileReaderMock),
    });

    // Trigger file selection
    fireEvent.change(fileInput, {
      target: { files: [file] },
    });

    // Ensure FileReader's readAsDataURL is called with the file
    expect(fileReaderMock.readAsDataURL).toHaveBeenCalledWith(file);

    // Simulate the onloadend event after the file is read
    fileReaderMock.onloadend();

    // Wait for the file preview to update (findByAltText checks if the image is rendered)
    const filePreviewImage = (await findByAltText(
      "Uploaded Preview"
    )) as HTMLInputElement;

    // Ensure that the file preview has been set with the mock result
    expect(filePreviewImage.src).toBe(fileReaderMock.result);
  });
});
