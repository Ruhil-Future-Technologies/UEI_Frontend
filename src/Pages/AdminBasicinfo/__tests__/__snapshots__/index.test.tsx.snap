// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Basic info Component renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row d-flex"
    >
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <label
          class="col-form-label"
        >
          First Name 
          <span>
            *
          </span>
        </label>
        <input
          autocomplete="off"
          class="form-control"
          data-testid="first_name"
          name="first_name"
          required=""
          type="text"
          value=""
        />
        <div />
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <label
          class="col-form-label"
        >
          Last Name 
          <span>
            *
          </span>
        </label>
        <input
          autocomplete="off"
          class="form-control"
          data-testid="last_name"
          name="last_name"
          required=""
          type="text"
          value=""
        />
        <div />
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          data-testid="gender"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            id="demo-row-radio-buttons-group-label"
          >
            Gender 
            <span>
              *
            </span>
          </label>
          <div
            class="MuiFormGroup-root MuiFormGroup-row MuiRadioGroup-root MuiRadioGroup-row css-qfz70r-MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary radiobutton css-i1zs00-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="gender"
                  type="radio"
                  value="male"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Male
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary radiobutton css-i1zs00-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  name="gender"
                  type="radio"
                  value="female"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Female
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 profiletext css-ahj2mt-MuiTypography-root"
        >
          Date of Birth 
          <span>
            *
          </span>
        </p>
        <div
          class="MuiBox-root css-1v5z18m"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-14ezuf6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                aria-label="datepicker_label"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                inputmode="text"
                maxlength="10"
                placeholder="DD/MM/YYYY"
                type="text"
                value="DD/MM/YYYY"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="Choose date"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <label
          class="col-form-label"
        >
          Father Name 
          <span>
            *
          </span>
        </label>
        <input
          autocomplete="off"
          class="form-control"
          data-testid="father_name"
          name="father_name"
          type="text"
          value=""
        />
        <div />
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <label
          class="col-form-label"
        >
          Mother Name 
          <span>
            *
          </span>
        </label>
        <input
          autocomplete="off"
          class="form-control"
          data-testid="mother_name"
          name="mother_name"
          type="text"
          value=""
        />
        <div />
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <label
          class="col-form-label"
        >
          Guardian Name
        </label>
        <input
          autocomplete="off"
          class="form-control"
          data-testid="guardian_name"
          name="guardian_name"
          type="text"
          value=""
        />
        <div />
      </div>
      <div
        class="col-md-6 pb-3 form_field_wrapper"
      >
        <div
          class="MuiFormControl-root css-1rxvxho-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="demo-select-small-label"
          >
            Department Name * 
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-crb59f-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-controls=":r3:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-select-small-label demo-select-small"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-select-small"
              role="combobox"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              data-testid="department_name"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Department name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div />
      </div>
      <div
        class="col-lg-12"
      >
        <div
          class="d-flex flex-wrap align-items-center gap-1"
        >
          <div
            class="image-container"
          >
            <div
              class="image-box"
            >
              <input
                class="image-checkbox"
                type="checkbox"
              />
              <img
                alt="male"
                src="male.png"
              />
              <span
                class="check-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m4.59-12.42L10 14.17l-2.59-2.58L6 13l4 4 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <label
            for="file"
          >
            <div
              class="upload-profile-image"
              role="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="UploadOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16h6v-6h4l-7-7-7 7h4zm3-10.17L14.17 8H13v6h-2V8H9.83zM5 18h14v2H5z"
                />
              </svg>
              <input
                accept="image/*"
                data-testid="profile_image"
                id="file"
                name="pic_path"
                style="display: none;"
                type="file"
              />
              Upload Your Picture
            </div>
          </label>
        </div>
      </div>
      <div
        class="col-lg-12"
      >
        <button
          class="btn btn-dark px-lg-5 mt-3 ms-auto d-block rounded-pill next-btn px-4"
          data-testid="next_button"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
